### ISSUE TITLE
Type conversions dont seem to work.

### ISSUE THREAD

**Issue by ericnewton76** [kahanu/System.Linq.Dynamic#92]
Given a test obj:
```csharp
public class TestObj
{
    int Id { get; set; }
}
```

if you try to build the expression with string values, the type conversion fails right away:

```csharp
using System.Linq.Dynamic;
//...
var list = new List<TestObj>() { new TestObj() { Id = 1 }, new TestObj() { Id = 2 } };

var __1 = list.Where("Id = @0", "1").ToList();  //fails to convert string "1" to integer 1.
var __2 = list.Where("Id = @0", 1).ToList(); //works because integer 1 pushed into parms object[]
```

**Comment by StefH** [comment]
This does work in https://github.com/StefH/System.Linq.Dynamic.Core

**Comment by ericnewton76** [comment]
Indeed, I'll close this if I get it working with that library instead.

**Comment by StefH** [comment]
@ericnewton76 Any luck ?

**Comment by ericnewton76** [comment]
I don't remember what I ended up doing.  I think it's still like that, so it'll probably fail. 

I'm using it to try to build queries via REST api's and passing the query almost directly into LINQ Dynamic: /api/endpoint?fieldname[eq]=value&integerfield[greaterThan]=10

I believe right now the integerField will fail because of the need to convert string "10" to int 10.

I'll check sometime this week.

**Comment by StefH** [comment]
In your scenario, using OData is also an option, it has parsing query parameters build in.
